models/issue_comment.go

```diff
@@ -1163,11 +1163,11 @@ func deleteComment(e db.Engine, comment *Comment) error {
 	}
 
 	if comment.Type == CommentTypeComment {
-		if _, err := e.Exec("UPDATE `issue` SET num_comments = num_comments - 1 WHERE id = ?", comment.IssueID); err != nil {
+		if _, err := e.ID(comment.IssueID).Decr("num_comments").Update(new(Issue)); err != nil {
 			return err
 		}
 	}
-	if _, err := e.Where("comment_id = ?", comment.ID).Cols("is_deleted").Update(&Action{IsDeleted: true}); err != nil {
+	if _, err := e.Where("comment_id = ?", comment.ID).Delete(&Action{}); err != nil {
```

Comment:
`Delete()` use under the hood the `DELETE` function, so it will remove the whole row from the database. Other than this here I think you can remove the whole Action.IsDeleted field(and add a migration to Delete those rows with `is_deleted = true`.

Reply:
We don't delete it but update `is_deleted` because contributions graph depends on it. This should not be changed.

Reply:
If some content is removed, it was supposed to never have been done. I think it should be completely removed everywhere.

If a user wants to get rid of some content, but the action table still keeps an excerpt of the original content, it is not removed and thus violates most privacy protection regulations I know of.

Please note that when removing an issue, I also deleted the actions in my recent PR. On repo deletion, actions are also already dropped. Removing it for comments, too, makes the behaviour more consistent.

Reply:
If we want to change the design, we need to change all the situations to keep consistent but only change 1 place.

Reply:
Sure, Gusted already mentioned we have to make it consistent - either by reverting the recent changes or by dropping the is_deleted column and removing the content.

I'd opt for dropping the Is_deleted column and the migration. I already removed actions when removing issues, also when removing a whole repo, the actions are dropped. As far as I can see, there are few cases left (like here in the issue comment removal).

Reply:
> I do not agree that it should be dropped, imho it is still shown in comment history to indicate that there was deleted record

Then it should have it's own comment type.

Reply:
Deleted comments are currently not displayed in the UI. Also, it does not make any sense to preserve information that is not used. And as said, it likely violates legal restrictions for everyone hosting Gitea if data is kept, but the user is suggested it is removed now.

If you really want to keep a record that something was removed, then you should - separately - introduce a record that only stores the data that something was removed, excluding any personally identifiable information.